src/main/java/uk/gov/justice/hmpps/casenotes/controllers/CaseNoteController.java

```diff
@@ -114,4 +114,59 @@ public CaseNote amendCaseNote(
         return caseNoteService.getUserCaseNoteTypes();
 
     }
+
+    @PostMapping(value = "/types", consumes = "application/json")
+    @ResponseStatus(HttpStatus.CREATED)
+    @ApiOperation(value = "Add New Case Note Type",
+            response = NewCaseNoteType.class,
+            notes = "Creates a new case note type")
+    @ApiResponses(value = {
+            @ApiResponse(code = 201, message = "The Case Note Type has been recorded. The updated object is returned including the status.", response = CaseNoteType.class),
+            @ApiResponse(code = 409, message = "The case note type has already been recorded. The current unmodified object (including status) is returned.", response = ErrorResponse.class)})
+    public CaseNoteType createCaseNoteType(@RequestBody @NotNull final NewCaseNoteType body) {
+        return caseNoteService.createCaseNoteType(body);
+    }
+
+    @PostMapping(value = "/types/{parentType}", consumes = "application/json")
+    @ResponseStatus(HttpStatus.CREATED)
+    @ApiOperation(value = "Add New Case Note Sub Type",
+            response = NewCaseNoteType.class,
+            notes = "Creates a new case note sub type")
+    @ApiResponses(value = {
+            @ApiResponse(code = 201, message = "The Case Note Sub Type has been recorded. The updated object is returned including the status.", response = CaseNoteType.class),
+            @ApiResponse(code = 409, message = "The case note sub type has already been recorded. The current unmodified object (including status) is returned.", response = ErrorResponse.class)})
+    public CaseNoteType createCaseNoteSubType(
+            @ApiParam(value = "Parent Case Note Type", required = true, example = "GEN") @PathVariable("parentType") final String parentType,
+            @RequestBody @NotNull final NewCaseNoteType body) {
+        return caseNoteService.createCaseNoteSubType(parentType, body);
+    }
+
+    @PutMapping(value = "/types/{parentType}", consumes = "application/json")
+    @ResponseStatus(HttpStatus.CREATED)
+    @ApiOperation(value = "Add New Case Note Type",
```

Comment:
`Update New Case Not Type` also Response status = 200 ?

Reply:
Think the value and notes still need to be updated?:
``` @ApiOperation(value = "Add New Case Note Sub Type",
            response = UpdateCaseNoteType.class,
            notes = "Creates a new case note sub type")
```
and below too

Reply:
Duh!